pydarn/utils/conversions.py

```diff
@@ -0,0 +1,124 @@
+DMAP_FORMAT_TYPES = {'c': 1,  # character (char)
+                     'h': 2,  # short integer
+                     'i': 3,  # integer (int)
+                     'f': 4,  # float
+                     'd': 8,  # double
+                     's': 9,  # string
+                     'q': 10,  # Long int
+                     'B': 16,  # unsigned char
+                     'H': 17,  # unsigned short int
+                     'I': 18,  # unsigned int
+                     'Q': 19}  # unsigned long int
+
+# python possible types that could be the python format type characters
+FORMAT_CONVERSION = {chr: 'c',  # Python doesn't actually have chars :/ they are just casted to strings
+                     np.char: 'c',  # not sure how to test this, you cannot use np.char() ¯\_(ツ)_/¯
+                     np.int8: 'c',  # not sure how a int is a char?
+                     np.int16: 'h',
+                     int: 'i',
+                     np.int32: 'i',
+                     float: 'f',
+                     np.float32: 'f',
+                     np.float64: 'd',
+                     str: 's',
+                     np.str_: 's',
+                     np.str: 's',
+                     np.int64: 'q',
+                     np.uint8: 'B',
+                     np.uint16: 'H',
+                     np.uint32: 'I',
+                     np.uint64: 'Q'}
+
+# !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+#       WARNING: This converts on best effort
+#       python does not have character types, if
+#       a string has a length of 1 then it will
+#       treated as a char.
+# !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+def dict2dmap(dmap_list: List[dict]) -> List[dict]:
+    """
+    This method converts list of dictionaries contain keys representing a field
+    name for the DMAP format and the item being the value of the field.
+
+    Parameters
+    ----------
+    dmap_list : List[dict]
+        List of dictionaries containing the field names and the
+        values of the data
+
+    Return
+    ------
+    dmap_records : List[dict]
+        List of Ordered dictionaries containing dmap data structures
+        DmapScalars and DmapArrays.
+    """
+    dmap_records = []
+    for dmap_dict in dmap_list:
+        dmap_record = OrderedDict()
+        for field, value in dmap_dict.items():
+            # if the value is an array then it is a DmapArray
+            if isinstance(value, list):
+                value = np.array(value)  # keep data structures consistent
+            if isinstance(value, np.ndarray):
+                shape = np.shape(value)
+                dimension = np.ndim(value)
+                # I did not implement the string length check to see if
+                # it is a list of chars because one would need to look at
+                # every element in the array, and it might be better to just
+                # store it as a string at this point.
```

Comment:
Nevermind the character encoding issues related to unicode vs. ascii, etc. Definitely something to revisit later, not right now.